Code/Framework/AzCore/AzCore/Name/NameDictionary.cpp

```diff
@@ -46,21 +47,34 @@ namespace AZ
         using namespace NameDictionaryInternal;
 
         AZ_Assert(s_instance, "NameDictionary not created!");
+        s_instance->UnloadDeferredNames();
```

Comment:
Why not invoke this function in the destructor?

This breaks the symmetry with the NameDictionary constructor loading the DeferredNames

Reply:
Oh I need to document this - the `NameData` destructor requires `s_instance` to be set, so calling this at destroy time is fatal.

Reply:
Ok thanks for the clarification.

In that case can move the Linking of Deferred Names to the NameDictionary::Create() function instead.

Reply:
👍 done